api/_dexes/cross-swap-service.ts

```diff
@@ -1513,49 +1554,114 @@ function _prepCrossSwapQuotesRetrievalA2A(
   });
 }
 
-function getMatchingStrategy(
-  baseStrategy: QuoteFetchStrategy,
-  otherStrategies: QuoteFetchStrategy[],
-  strictTradeType: boolean
-) {
-  // If strict trade type is enabled, we need to check if the matching strategy supports
-  // selling the entire balance.
-  return strictTradeType
-    ? // First check if other strategy with same name supports selling the entire balance.
-      (otherStrategies.find((otherStrategy) => {
-        const sameStrategy =
-          otherStrategy.strategyName === baseStrategy.strategyName;
-        let supportsSellEntireBalance = false;
-        try {
-          otherStrategy.assertSellEntireBalanceSupported();
-          supportsSellEntireBalance = true;
-        } catch (error) {
-          supportsSellEntireBalance = false;
-        }
-        return sameStrategy && supportsSellEntireBalance;
-      }) ??
-        // If no other strategy with same name supports selling the entire balance,
-        // check if any other strategy supports selling the entire balance.
-        otherStrategies.find((otherStrategy) => {
-          try {
-            otherStrategy.assertSellEntireBalanceSupported();
-            return true;
-          } catch (error) {
-            return false;
-          }
-        }) ??
-        // If no other strategy supports selling the entire balance,
-        // return the first other strategy with same name.
-        otherStrategies.find((otherStrategy) => {
-          return otherStrategy.strategyName === baseStrategy.strategyName;
-        }) ??
-        // If no other strategy supports selling the entire balance,
-        // return the first other strategy.
-        otherStrategies[0])
-    : // If strict trade type is disabled, return the strategy with same name.
-      (otherStrategies.find((otherStrategy) => {
-        return otherStrategy.strategyName === baseStrategy.strategyName;
-      }) ?? otherStrategies[0]);
+export function getMatchingStrategy(params: {
+  baseStrategy: QuoteFetchStrategy;
+  otherStrategies: QuoteFetchStrategy[];
+  strictTradeType: boolean;
+  includeSources?: string[];
+  chainId: number;
+}) {
+  const {
+    baseStrategy,
+    otherStrategies,
+    strictTradeType,
+    includeSources,
+    chainId,
+  } = params;
+
+  const baseStrategyHasSources = hasSources(baseStrategy, chainId, {
+    includeSources,
+  });
+
+  // Helper to check if strategy matches source requirements
+  const hasValidSources = (strategy: QuoteFetchStrategy): boolean => {
+    const strategyHasSources = hasSources(strategy, chainId, {
+      includeSources,
+    });
+    return baseStrategyHasSources
+      ? baseStrategyHasSources && strategyHasSources
+      : strategyHasSources;
```

Comment:
Hmm won't this always just be `strategyHasSources`?
```suggestion
    return strategyHasSources;
```